lisa/features/gpu.py

```diff
@@ -257,16 +265,18 @@ def _install_cuda_driver(self, version: str) -> None:
                         f"failed to add repo {repo_entry}"
                     ),
                 )
-                # the latest version cuda-drivers-510 has issues
-                # nvidia-smi
-                # No devices were found
-                # dmesg
-                # NVRM: GPU 0001:00:00.0: RmInitAdapter failed! (0x63:0x55:2344)
-                # NVRM: GPU 0001:00:00.0: rm_init_adapter failed, device minor number 0
-                cuda_package_versions = {"1804": "530"}
-                # 545 is the latest version available for Ubuntu 2004+
-                package_version = cuda_package_versions.get(release, "545")
-                self._node.os.install_packages(f"cuda-drivers-{package_version}")
+                result = self._node.execute(
+                    f"apt search {cuda_package_name}", sudo=True
+                )
+                available_versions = cuda_drivers_package_pattern.findall(result.stdout)
+
+            if available_versions:
+                # Sort versions and select the highest one
+                highest_version = max(available_versions)
```

Comment:
for now this will work fine, but with double digits lexicographical max may not work the way we want, perhaps you can check this?

Reply:
ignore this, this should be fine

Reply:
The packages does not use semantic versioning. 
The examples of packages are "cuda-drivers-550", "cuda-drivers-555", "cuda-drivers-560" and so on

https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2404/x86_64/